nevergrad/optimization/optimizerlib.py

```diff
@@ -1109,4 +1164,74 @@ def __call__(self, instrumentation: Union[int, Instrumentation],
 MidQRBO = ParametrizedBO(initialization="Hammersley", middle_point=True).with_name("MidQRBO", register=True)
 LBO = ParametrizedBO(initialization="LHS").with_name("LBO", register=True)
 
+
+@registry.register
+class PBIL(base.Optimizer):
+    """
+    Implementation of the discrete algorithm PBIL
+
+    https://www.ri.cmu.edu/pub_files/pub1/baluja_shumeet_1994_2/baluja_shumeet_1994_2.pdf
+    """
+
+    # pylint: disable=too-many-instance-attributes
+
+    def __init__(self, instrumentation: Union[int, Instrumentation], budget: Optional[int] = None, num_workers: int = 1) -> None:
+        super().__init__(instrumentation, budget=budget, num_workers=num_workers)
+
+        num_categories = 2
+        self.p: np.ndarray = np.ones((1, self.dimension)) / num_categories
+        self.alpha = 0.3
+        self.llambda = max(100, num_workers)  # size of the population
+        self.mu = self.llambda // 2  # number of selected candidates
+        self._population: List[Tuple[float, np.ndarray]] = []
+
+    def _internal_ask_candidate(self) -> base.Candidate:
+        unif = self._rng.uniform(size=self.dimension)
+        data = (unif > self.p[0]).astype(float)
```

Comment:
Maybe we should check the compatibility with the instrumentation ?
Another possibility is to extend the algorithm to more instrumentations, but maybe later.